plugins/graphql-kotlin-gradle-plugin/src/main/kotlin/com/expediagroup/graphql/plugin/gradle/GraphQLPluginExtension.kt

```diff
@@ -62,7 +63,7 @@ open class GraphQLPluginClientExtension {
     /** Connect and read timeout configuration for executing introspection query/download schema */
     internal val timeoutConfig: TimeoutConfig = TimeoutConfig()
 
-    fun timeout(config: TimeoutConfig.() -> Unit = {}) {
-        timeoutConfig.apply(config)
+    fun timeout(action: Action<TimeoutConfig>) {
```

Comment:
there appears to be an issue with SAM conversion or maybe it is Groovy issue but I'm hitting a problem when attempting to invoke `Action` from another `Action`, i.e.

```groovy
graphql {
    client {
        endpoint = "http://localhost:8080/graphql"
        packageName = "com.example.generated"
        // above works fine, but complains on the below, looks like "this" from the action refers to outer one
        timeout {
            connect = 10000
            read = 30000
        }
    }
}
```

results in 

`Could not set unknown property 'connect' for object of type com.expediagroup.graphql.plugin.gradle.GraphQLPluginClientExtension.`

Reply:
Is there a solution for this? I'm trying to set a connection timeout value per the documentation [here](https://expediagroup.github.io/graphql-kotlin/docs/plugins/gradle-plugin-tasks/#extension) and getting

```
> Could not set unknown property 'connect' for object of type com.expediagroup.graphql.plugin.gradle.GraphQLPluginClientExtension.
```

Reply:
@krispharper - which version of the plugin are you using? this should be working fine now

Reply:
if you are using Groovy you might need to explicitly name the parameter as otherwise it might be confused on which object to apply those params.

```groovy
timeout { t ->
  t.connect = 10000
  t.read = 30000
}
```

Reply:
Got it, making that change worked. Thanks a bunch.